File: programme/fibonnaci-series-using-recursion/FibonacciRecursion.java
// Write a program to print the Fibonacci series for 'n' numbers
import java.util.Scanner;

class FibonacciRecursion {
    public static void main(String args[]) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter a number : ");
        int num = sc.nextInt();

        System.out.print("Fibonacci Series :");
        for (int i = 0; i <= num; i++) {
            System.out.print(" " + fibo(i));
        }
        System.out.println();
    }

    static int fibo(int num) {
        if (num < 2) {
            return num;
        }
        return fibo(num - 1) + fibo(num - 2);
    }
}

File: programme/fibonnaci-series-using-recursion/fibonacci_recursion.js
/*
Write a programmeto print Fibonacci series using recursion.

Example:
Input: n = 10.
Output: 0 1 1 2 3 5 8 13 21 34
*/

const fib = (n) => {
  if (n === 0) return 0;
  if (n <= 2) return 1;
  return fib(n - 1) + fib(n - 2);
};

const printFibs = (n) => {
  let str = "";
  for (let i = 0; i < n; i++) {
    str += `${fib(i)} `;
  }
  console.log(str);
};

printFibs(10);
